### PR TITLE
ROX-17931: Refactor deduping queue

### PR DESCRIPTION
**Pull request by lvalerom**
## Description

Follow up to #10269. Refactor the deduping queue into a generic queue.

## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

### Here I tell how I validated my change

- [x] CI

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: central/sensor/service/connection/worker_queue.go [hunk 1]
```diff
@@ -61,32 +65,37 @@ func (w *workerQueue) push(msg *central.MsgFromSensor) {
 		idx = w.indexFromKey(msg.HashKey)
 	}
 
-	w.queues[idx].push(msg)
+	w.queues[idx].Push(msg)
```
[the remark this answers is not in the document]
Comment (lvalerom) [reply central/sensor/service/connection/worker_queue.go:68]: Done

**Comment by lvalerom** [comment]
> I would maybe only suggest to add unit test coverage in pkg/dedupingqueue/deduping_queue.go for the lines that are uncovered and are not about metrics (there are really few such places).

@vikin91 I've added more coverage if you want to have another look 🙂

**Review by vikin91 — APPROVED** [review]
Thanks for improving!

**Pull request merged**